File: apps/admin-v2/src/domain/repositories/onboarding.repository.ts
import type { Tenant, BusinessType } from '../entities/tenant';

export interface RegisterTenantData {
  businessName: string;
  ownerName: string;
  email: string;
  password: string;
}

export interface OnboardingRepository {
  register(data: RegisterTenantData): Promise<{ token: string; tenant: Tenant }>;
  verify(code: string): Promise<void>;
  checkSlug(slug: string): Promise<{ available: boolean }>;
  setBusinessType(type: BusinessType, createServices: boolean): Promise<void>;
}

File: apps/admin_mobile/lib/domain/repositories/onboarding_repository.dart
import '../entities/tenant.dart';

abstract class OnboardingRepository {
  Future<({String token, Tenant tenant})> register({
    required String businessName,
    required String ownerName,
    required String email,
    required String password,
  });
  Future<void> verify(String code);
  Future<bool> checkSlug(String slug);
  Future<void> setBusinessType(String type, bool createServices);
}
